src/cargo/util/toml/mod.rs

```diff
@@ -2244,32 +2189,28 @@ impl<P: ResolveToPath + Clone> TomlDependency<P> {
                 optional,
             }) => {
                 cargo_features.require(Feature::workspace_inheritance())?;
-
-                let inherit = inherit().context(format!(
-                    "You cannot inherit fields from a parent workspace currently, tried to on `[dependency.{}]`",
-                    label
-                ))?;
-
-                get_ws(inherit).map_or(
-                    Err(anyhow!("error reading `{label}` from workspace root manifest's `[workspace.{label}]`")),
-                    |dep| {
-                        match dep {
-                            TomlDependency::Simple(s) => Ok(TomlDependency::Simple(s)),
-                            TomlDependency::Detailed(d) => {
-                                let mut dep = d.clone();
-                                dep.add_features(features);
-                                dep.update_optional(optional);
-                                Ok(TomlDependency::Detailed(dep))
-                            }
-                            TomlDependency::Workspace(_) => bail!("tried to inherit from `dependency.{label}`, but it also tried to inherit from a workspace")
+                get_ws_dependency().context(format!(
+                    "error reading `dependencies.{}` from workspace root manifest's `[workspace.dependencies.{}]`",
+                    label, label
+                )).map(|dep| {
+                    match dep {
+                        TomlDependency::Simple(s) => TomlDependency::Simple(s),
+                        TomlDependency::Detailed(d) => {
+                            let mut dep = d.clone();
+                            dep.add_features(features);
+                            dep.update_optional(optional);
+                            TomlDependency::Detailed(dep)
                         }
+                        // We check for this when we parse a toml file
+                        TomlDependency::Workspace(_) => unreachable!(),
```

Comment:
Please re-phrase the comment to be a parameter to the `unreachable!` macro, it accepts a value so we might as well use it and tell the user whats going on rather than having to dig in to find the comment for what invariant was violated.

Reply:
I updated this to be
```rust
TomlDependency::Workspace(_) => {
    unreachable!(
        "We check that no workspace defines dependencies with \
        `{{ workspace = true }}` when we read a manifest from a string. \
        this should not happen but did on {}",
        label
    )
},
```

I think it gives a better description of what is going on and where we check to make sure this never happens.